wallet-new/src/Cardano/Wallet/Kernel/Wallets.hs

```diff
@@ -142,3 +184,81 @@ createWalletHdRnd pw hasSpendingPassword name assuranceLevel esk utxo = do
             if hasSpendingPassword then HD.HasSpendingPassword created
                                    else HD.NoSpendingPassword
 
+deleteHdWallet :: PassiveWallet
+               -> HD.HdRootId
+               -> IO (Either HD.UnknownHdRoot ())
+deleteHdWallet wallet rootId = do
+    -- STEP 1: Remove the HdRoot via an acid-state transaction which will
+    --         also delete any associated accounts and addresses.
+    res <- update' (wallet ^. wallets) $ DeleteHdRoot rootId
+    case res of
+        Left err -> return (Left err)
+        Right () -> do
+            -- STEP 2: Purge the key from the keystore.
+            Keystore.delete (WalletIdHdRnd rootId) (wallet ^. walletKeystore)
```

Comment:
Do we worry about exceptions here? What if step 1 happens but step 2 does not?

Reply:
I thought about this, and it's not clear to me what's the best way to tackle this: if exceptions strikes _after_ we deleted the wallet from the DB, by all means we should ensure that the Keystore purges the key to avoid having dangling keys. Perhaps we should run the db update & keystore purging in a block with exceptions masked? 🤔